From aa18490ba5ce05a43b7046006ecc258633255c54 Mon Sep 17 00:00:00 2001
From: Henrik Grimler <grimler@termux.dev>
Date: Tue, 4 Mar 2025 11:00:35 +0100
Subject: [PATCH] libusb: build with iconv when __TERMUX__ is set

Android does not have libiconv, but termux does, so add some
__TERMUX__ checks.
---
 libusb/CMakeLists.txt | 4 ----
 libusb/hid.c          | 6 +++---
 2 files changed, 3 insertions(+), 7 deletions(-)

diff --git a/libusb/CMakeLists.txt b/libusb/CMakeLists.txt
index 6cd48c4f70f6..beba0a3fdb33 100644
--- a/libusb/CMakeLists.txt
+++ b/libusb/CMakeLists.txt
@@ -22,7 +22,6 @@ target_link_libraries(hidapi_libusb PRIVATE Threads::Threads)
 if(HIDAPI_NO_ICONV)
     target_compile_definitions(hidapi_libusb PRIVATE NO_ICONV)
 else()
-    if(NOT ANDROID)
         include(CheckCSourceCompiles)
 
         if(NOT CMAKE_VERSION VERSION_LESS 3.11)
@@ -77,9 +76,6 @@ else()
         if(HIDAPI_ICONV_CONST)
             target_compile_definitions(hidapi_libusb PRIVATE "ICONV_CONST=const")
         endif()
-    else()
-        # On Android Iconv is disabled on the code level anyway, so no issue;
-    endif()
 endif()
 
 set_target_properties(hidapi_libusb
diff --git a/libusb/hid.c b/libusb/hid.c
index 188e536d537d..d3861f772673 100644
--- a/libusb/hid.c
+++ b/libusb/hid.c
@@ -42,7 +42,7 @@
 
 /* GNU / LibUSB */
 #include <libusb.h>
-#if !defined(__ANDROID__) && !defined(NO_ICONV)
+#if (!defined(__ANDROID__) || defined(__TERMUX__)) && !defined(NO_ICONV)
 #include <iconv.h>
 #ifndef ICONV_CONST
 #define ICONV_CONST
@@ -406,7 +406,7 @@ static wchar_t *get_usb_string(libusb_device_handle *dev, uint8_t idx)
 	int len;
 	wchar_t *str = NULL;
 
-#if !defined(__ANDROID__) && !defined(NO_ICONV) /* we don't use iconv on Android, or when it is explicitly disabled */
+#if (!defined(__ANDROID__) || defined(__TERMUX__)) && !defined(NO_ICONV) /* we don't use iconv on non-Termux Android, or when it is explicitly disabled */
 	wchar_t wbuf[256];
 	/* iconv variables */
 	iconv_t ic;
@@ -432,7 +432,7 @@ static wchar_t *get_usb_string(libusb_device_handle *dev, uint8_t idx)
 	if (len < 2) /* we always skip first 2 bytes */
 		return NULL;
 
-#if defined(__ANDROID__) || defined(NO_ICONV)
+#if (defined(__ANDROID__) || defined(NO_ICONV)) && !defined(__TERMUX__)
 
 	/* Bionic does not have iconv support nor wcsdup() function, so it
 	   has to be done manually.  The following code will only work for
-- 
2.49.0

